src/password/scripts/directives.js

```diff
@@ -44,13 +45,9 @@
                                         }, function (response) {
                                             scope.errorData = response.data;
 
-                                            angular.forEach(response.data, function (error, field) {
-                                                error = angular.isArray(error) ? error[0] : error;
-                                                if (angular.isDefined(scope.fields[field])) {
-                                                    scope.fields[field].errors = error;
-                                                }
-                                                scope.errors[field] = error;
-                                            });
+                                            var errors = catchErrors.all(response, scope.fields);
+                                            angular.merge(scope.errors, errors.nonFieldErrors);
```

Comment:
It would seem simpler to do the `angular.merge` in `catchErrors` (as the errors are parsed).